packages/integrations/src/openmediavault/openmediavault-integration.ts

```diff
@@ -0,0 +1,151 @@
+import { Integration } from "../base/integration";
+import { IntegrationTestConnectionError } from "../base/test-connection-error";
+import type { HealthMonitoring } from "../types";
+import { cpuTempSchema, fileSystemSchema, smartSchema, systemInformationSchema } from "./openmediavault-types";
+
+export class OpenMediaVaultIntegration extends Integration {
+  private async makeOpenMediaVaultRPCCallAsync(
+    serviceName: string,
+    method: string,
+    params: Record<string, unknown>,
+    headers: Record<string, string> = {},
+  ): Promise<Response> {
+    return await fetch(`${this.integration.url}/rpc.php`, {
+      method: "POST",
+      headers: {
+        "Content-Type": "application/json",
+        ...headers,
+      },
+      body: JSON.stringify({
+        service: serviceName,
+        method,
+        params,
+      }),
+    });
+  }
+
+  private async omvAuthAsync() {
```

Comment:
Is it really required to call this function every request?
You will create double the noise for every API call. It isn't a bad thing but would be nice to avoid unnessesary network traffic if possible. Does the Cookie have an expiry?
If yes, maybe you can only perform this logic if the Cookie expired or will expire soon?
@Meierschlumpf I think your guidance would be nice here. Would a cache make sense here? And if yes, how do we deal with integration updates (eg. change URL)?

Reply:
I found a nice solution.